keel-intent-aws/src/main/kotlin/com/netflix/spinnaker/keel/intent/aws/loadbalancer/ClassicLoadBalancerConverter.kt

```diff
@@ -0,0 +1,131 @@
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.keel.intent.aws.loadbalancer
+
+import com.amazonaws.services.elasticloadbalancing.model.HealthCheck
+import com.amazonaws.services.elasticloadbalancing.model.Listener
+import com.amazonaws.services.elasticloadbalancing.model.ListenerDescription
+import com.amazonaws.services.elasticloadbalancing.model.LoadBalancerDescription
+import com.netflix.spinnaker.keel.clouddriver.CloudDriverCache
+import com.netflix.spinnaker.keel.dryrun.ChangeSummary
+import com.netflix.spinnaker.keel.intent.aws.loadbalancer.AvailabilityZoneConfig.Automatic
+import com.netflix.spinnaker.keel.intent.aws.loadbalancer.AvailabilityZoneConfig.Manual
+import com.netflix.spinnaker.keel.intent.processor.converter.SpecConverter
+import com.netflix.spinnaker.keel.model.Job
+import org.springframework.stereotype.Component
+
+@Component
+class ClassicLoadBalancerConverter(
+  private val cloudDriver: CloudDriverCache
+) : SpecConverter<ClassicLoadBalancerSpec, LoadBalancerDescription> {
+  override fun convertToState(spec: ClassicLoadBalancerSpec): LoadBalancerDescription {
+    val vpcId = cloudDriver.networkBy(spec.vpcName!!, spec.accountName, spec.region).id
+    val zones = cloudDriver.availabilityZonesBy(spec.accountName, vpcId, spec.region)
+
+    return LoadBalancerDescription()
+      .withLoadBalancerName(spec.name)
+      .withScheme(spec.scheme?.toString())
+      .withVPCId(vpcId)
+      .withAvailabilityZones(
+        spec.availabilityZones.let { zoneConfig ->
+          when (zoneConfig) {
+            is Manual -> zoneConfig.availabilityZones
+            else -> zones
+          }
+        }
+      )
+      .withSecurityGroups(spec.securityGroupNames)
+      .withHealthCheck(
+        spec.healthCheck.run {
+          HealthCheck()
+            .withHealthyThreshold(healthyThreshold)
+            .withUnhealthyThreshold(unhealthyThreshold)
+            .withInterval(interval)
+            .withTarget(target.toString())
+            .withTimeout(timeout)
+        }
+      )
+      .withListenerDescriptions(
+        spec.listeners.map {
+          ListenerDescription()
+            .withListener(
+              Listener()
+                .withLoadBalancerPort(it.loadBalancerPort)
+                .withProtocol(it.protocol.toString())
+                .withInstancePort(it.instancePort)
+                .withInstanceProtocol(it.instanceProtocol.toString())
+                .withSSLCertificateId(it.sslCertificateId)
+            )
+        }
+      )
+  }
+
+  override fun convertFromState(state: LoadBalancerDescription): ClassicLoadBalancerSpec =
+    state.run {
+      val vpc = cloudDriver.networkBy(vpcId!!)
```

Comment:
I don't think this network logic is totally correct (this existed previous) so I'll need to revisit and ensure we do the right things with vpc / subnet names.